actors/runtime/exitcode/reserved.go

```diff
@@ -0,0 +1,65 @@
+package exitcode
+
+type ExitCode int64
+
+func (x ExitCode) IsSuccess() bool {
+	return x == Ok
+}
+
+func (x ExitCode) IsError() bool {
+	return !x.IsSuccess()
+}
+
+const (
+	Ok = ExitCode(0)
+	// NOTE: none of the system error codes should ever be used by actors, so we could move their definition
+	// out of this package and into the runtime spec.
+
+	// Indicates failure to find an actor in the state tree.
+	SysErrActorNotFound = ExitCode(1)
+
+	// Indicates failure to find the code for an actor.
+	SysErrActorCodeNotFound = ExitCode(2)
+
+	// Indicates failure to find a method in an actor.
+	SysErrInvalidMethod = ExitCode(3)
+
+	// Indicates syntactically invalid parameters for a method.
+	SysErrInvalidParameters = ExitCode(4)
+
+	// Indicates a message sender has insufficient funds for a message's execution.
+	SysErrInsufficientFunds = ExitCode(5)
+
+	// Indicates a message invocation out of sequence.
+	SysErrInvalidCallSeqNum = ExitCode(6)
+
+	// Indicates message execution (including subcalls) used more gas than the specified limit.
+	SysErrOutOfGas = ExitCode(7)
+
+	// Indicates a message execution is forbidden for the caller.
+	SysErrForbidden = ExitCode(8)
+
+	// Indicates actor code performed a disallowed operation. Disallowed operations include:
+	// - mutating state outside of a state acquisition block
+	// - failing to invoke caller validation
+	// - aborting with a reserved exit code (including success or a system error).
+	SysErrorIllegalActor = ExitCode(9)
+
+	// Indicates an invalid argument passed to a runtime method.
+	SysErrorIllegalArgument = ExitCode(10)
+
+	// MethodSubcallError is returned when an actor method's Send call has
+	// returned with a failure error code (and the Send call did not specify
+	// to ignore errors).
+	SysErrorSubcallAbort = ExitCode(11)
+
+	// Reserved exit codes, do not use.
+	SysErrorReserved1 = ExitCode(12)
+	SysErrorReserved2 = ExitCode(13)
+	SysErrorReserved3 = ExitCode(14)
+	SysErrorReserved4 = ExitCode(15)
+)
+
```

Comment:
do we not need `RuntimeAssertFailure` because `Assert` will just halt the VM and not return?

Reply:
It was un-used, and yes assert failures don't go on chain.